### PR TITLE
fix sparse_softmax_cross_entropy bug when logits is negative infinity

### PR DESCRIPTION
**Pull request by liangkaihuan**
When all elements of logits are negative infinity, sparse_softmax_cross_entropy will output NAN. Sub the maximum first can fix the problem.

### CODE REVIEW

**Review by liangkaihuan — COMMENTED** [review]
sub the maxmium

**Comment by TomWildenhain-Microsoft** [comment]
Thanks so much for the contribution!  To confirm, you are avoiding the negative infinities entirely by shrinking the inputs to the exp.  Is that correct?

**Review by TomWildenhain-Microsoft — APPROVED** [review]
